### ISSUE TITLE
[Critical] Slackin will break in February 2021

### ISSUE THREAD

**Issue by mrfrase3** [rauchg/slackin#417]
Whilst looking into the api calls that slackin uses, I noticed it is using `channels.list`, which is deprecated and noted to stop working in February 2021. From my understanding of the code, all running slackin instances will stop running, erroring whenever being initialised.

https://api.slack.com/methods/channels.list

Although, with slack no longer providing a way to generate legacy tokens, this further complicates the future of slackin.

N.B. [slackin-extended](https://github.com/emedvedev/slackin-extended) has [(unintentionally?)](https://github.com/emedvedev/slackin-extended/pull/108) resolved this issue back in January, version 2.0.0. Although it still has the same issue with legacy tokens.

**Comment by jbw976** [comment]
We've been using slackin untouched for quite awhile now at https://slack.crossplane.io/ and https://slack.rook.io/ and just today we are seeing errors/crashes.  From the investigation so far, I'm wondering if this issue here about `channels.list` has finally caught up with us.  Anyone else just seeing their slackin break today? 🤔 

https://github.com/crossplane/crossplane/issues/3872#issuecomment-1470964839

**Comment by JoeDuncko** [comment]
> We've been using slackin untouched for quite awhile now at https://slack.crossplane.io/ and https://slack.rook.io/ and just today we are seeing errors/crashes. From the investigation so far, I'm wondering if this issue here about `channels.list` has finally caught up with us. Anyone else just seeing their slackin break today? 🤔
> 
> [crossplane/crossplane#3872 (comment)](https://github.com/crossplane/crossplane/issues/3872#issuecomment-1470964839)

Yup! I too am here because we realized our Slackin ( http://slack.codeyoungstown.com/ ) broke yesterday. Not sure what's going on yet.

Edit:

Switching to Slack Invite Links, see https://github.com/rauchg/slackin/issues/416#issuecomment-678851251

**Comment by mrfrase3** [comment]
Warning, rant inbound:

I wouldn't be surprised if slack has been going around and conveniently deprecating APIs that open communities use. From what I've gleened back when I looked into this, slack only cares about its enterprise users and this will only get worse since the salesforce purchase.

The assessment I made was that sooner or later, a migration to discord would become inevitable, where they actively support open communities and provide them the tooling they need. You can turn your discord server into a community server for free, and that unlocks a bunch of tooling, including forums which are amazing for busy channels like help.

To help with migrations, and a bit of a middle finger to slack, I made a slack export viewer called [slax.pro](https://slax.pro) that allows you to view the entire history of your public channels which slack allows you to export for free.

If you're really devoted to slack, you can try slackin-extended that is more actively developed, or you can manually update the invite like every 30 days, or you can write a webscraper to auto login and update the link, which I have done in the past with puppeteer.

**Comment by daveyarwood** [comment]
> We've been using slackin untouched for quite awhile now at https://slack.crossplane.io/ and https://slack.rook.io/ and just today we are seeing errors/crashes. From the investigation so far, I'm wondering if this issue here about `channels.list` has finally caught up with us. Anyone else just seeing their slackin break today? :thinking: 
> 
> [crossplane/crossplane#3872 (comment)](https://github.com/crossplane/crossplane/issues/3872#issuecomment-1470964839)

Same here. https://slack.alda.io was working fine untouched for years, but it just randomly started crashing last week. I'm going to look into some of the alternatives now, as it looks like Slackin is no longer maintained.

EDIT: My current plan is to periodically generate a new invite link, and host a simple webpage at https://slack.alda.io (since there are already a lot of links to that URL) that redirects to the current invite link.

**Comment by jbw976** [comment]
@daveyarwood that's exactly what we've done for both slack.crossplane.io and slack.rook.io - a simple static website that just redirects to a Slack workspace invitation link.  it seems to be working fine so far 🙌 🙏 

Feel free to take a look at those repos for any hints that may be useful for you 😉 

* https://github.com/crossplane/slack.crossplane.io
* https://github.com/rook/slack.rook.io

**Comment by simi** [comment]
> @daveyarwood that's exactly what we've done for both slack.crossplane.io and slack.rook.io - a simple static website that just redirects to a Slack workspace invitation link. it seems to be working fine so far raised_hands pray
> 
> Feel free to take a look at those repos for any hints that may be useful for you wink
> 
> * https://github.com/crossplane/slack.crossplane.io
> * https://github.com/rook/slack.rook.io

@jbw976 And what are you going to do in next 30 days once the invite link expires? Update manually?

**Comment by simi** [comment]
Ahh, it seems possible to create never expiring link using link settings.

![image](https://user-images.githubusercontent.com/193936/227194300-48bb7cdf-4dcc-4da8-820c-ed1717ef53a5.png)
![image](https://user-images.githubusercontent.com/193936/227194333-0650984d-8e3f-4404-91e2-08ac7791936e.png)

**Comment by daveyarwood** [comment]
That's true, but it also says that a link can be shared with up to 400 people. So I think we will still need to manually update our links periodically whenever we reach that limit.

At my scale, that doesn't bother me too much, but I could definitely see that being problematic if you have a rapidly growing community.
